src/openms/source/ANALYSIS/TOPDOWN/PeakGroup.cpp

```diff
@@ -313,41 +324,79 @@ namespace OpenMS
       int c_zero = 0;
       for (int z = max_sig_charge; z <= t_nmax_abs_charge; z++)
       {
-        if (per_charge_int_[z] <= 0)
+        if (per_charge_signal_pwr_[z] <= 0)
         {
           c_zero++;
         }
         else
         {
           c_zero = 0;
+          nmax_abs_charge = z;
         }
         if (c_zero > 2)
         {
           break;
         }
-        nmax_abs_charge = z;
+
       }
       c_zero = 0;
       for (int z = max_sig_charge; z >= t_nmin_abs_charge; z--)
       {
-        if (per_charge_int_[z] <= 0)
+        if (per_charge_signal_pwr_[z] <= 0)
         {
           c_zero++;
         }
         else
         {
           c_zero = 0;
+          nmin_abs_charge = z;
         }
         if (c_zero > 2)
         {
           break;
         }
-        nmin_abs_charge = z;
       }
     }
+    if(max_abs_charge_ != nmax_abs_charge || min_abs_charge_ != nmin_abs_charge)
+    {
+      std::vector<LogMzPeak> new_sig;
+      std::vector<LogMzPeak> new_noise;
+      new_sig.reserve(size());
+      new_noise.reserve(noisy_peaks_.size());
+
+      for(auto p:logMzpeaks_)
+      {
+        if(p.abs_charge < nmin_abs_charge || p.abs_charge > nmax_abs_charge)
+        {
+          continue;
+        }
+        new_sig.push_back(p);
+      }
+
+      for(auto p:noisy_peaks_)
+      {
+        if(p.abs_charge < nmin_abs_charge || p.abs_charge > nmax_abs_charge)
+        {
+          continue;
+        }
+        new_noise.push_back(p);
+      }
+
+      std::vector<LogMzPeak>().swap(noisy_peaks_);
+      std::vector<LogMzPeak>().swap(logMzpeaks_);
+      new_noise.swap(noisy_peaks_);
+      new_sig.swap(logMzpeaks_);
```

Comment:
This part doesn't make sense to me. Again, no comments to understand intention here (e.g., I assume there is some memory bottleneck).
For me it looks like e.g. new_noise get's destroyed in line 392 anyway.
So why not do a noisy_peaks_ = std::move(new_noise) (or just the swap).
Maybe I missed something but hard to tell without comments

Reply:
This has been removed!